raft/raft.go

```diff
@@ -370,10 +378,12 @@ func (r *raft) sendHeartbeat(to uint64) {
 	// an unmatched index.
 	commit := min(r.prs[to].Match, r.raftLog.committed)
 	m := pb.Message{
-		To:     to,
-		Type:   pb.MsgHeartbeat,
-		Commit: commit,
+		To:      to,
+		Type:    pb.MsgHeartbeat,
+		Commit:  commit,
+		Context: ctx,
```

Comment:
Do we need to send the user-supplied context on the wire with the heartbeat? I know we need to use the context to connect ReadIndex and the result in Ready, but I think for the heartbeat we can express this in terms of the term and index instead of a user-supplied value. This would allow multiple reads to piggyback on the same message.